### ISSUE TITLE
upper_case_acronyms doesn't check the enum name itself.

### ISSUE THREAD

**Issue by TennyZhuang** [rust-lang/rust-clippy#9579]
### Summary

upper_case_acronyms only check the enum items without the enum name itself.

### Lint Name

upper_case_acronyms

### Reproducer

I tried this code:

```rust
enum HTTPResponse {
    A,
    B,
}

fn main() {
    let _ = HTTPResponse::A;
}
```

I expected to see this happen:

Warnings like struct

Instead, this happened:

check passed

### Version

```text
rustc 1.65.0-nightly (02654a084 2022-08-30)
binary: rustc
commit-hash: 02654a0844f5c8d29bac318c3c6c666da3d8543d
commit-date: 2022-08-30
host: aarch64-apple-darwin
release: 1.65.0-nightly
LLVM version: 15.0.0
```
